app/src/main/java/com/nononsenseapps/feeder/model/FeedParser.kt

```diff
@@ -204,6 +235,132 @@ class FeedParser(override val di: DI) : DIAware {
+                        it.size < 4
+                    }.orEmpty()
+                    .ifEmpty { getUserPublishRelays(publicKey) }
+            logDebug(LOG_TAG, "Relays from Nip19 -> ${relayList.joinToString(separator = ", ")}")
+            relayList
+                .ifEmpty { defaultFetchRelays }
+                .forEach { relayUrl ->
+                    nostrClient.addReadRelay(relayUrl)
+                }
+            nostrClient.connect()
+            val profileInfo =
+                nostrClient.fetchMetadata(
+                    publicKey = publicKey,
+                    timeout = Duration.ofSeconds(5L),
+                )
+            logDebug(LOG_TAG, profileInfo.asPrettyJson())
+
+            // Check if all relays in relaylist can be connected to
+            AuthorNostrData(
+                uri = possibleNostrProfile.toNostrUri(),
+                name = profileInfo.getName().toString(),
+                publicKey = publicKey,
+                imageUrl = profileInfo.getPicture().toString(),
+                relayList = nostrClient.relays().map { relayEntry -> relayEntry.key },
+            )
+        }
+    }
+
+    private suspend fun getUserPublishRelays(userPubkey: PublicKey): List<String> {
+        val userRelaysFilter =
+            Filter()
+                .author(userPubkey)
+                .kind(
+                    Kind.fromEnum(KindEnum.RelayList),
+                )
+
+        nostrClient.removeAllRelays()
+        defaultMetadataRelays.forEach { relayUrl ->
+            nostrClient.addReadRelay(relayUrl)
+        }
+        nostrClient.connect()
+        val potentialUserRelays =
+            nostrClient.fetchEventsFrom(
+                urls = defaultMetadataRelays,
+                filters = listOf(userRelaysFilter),
+                timeout = Duration.ofSeconds(5),
+            )
+        val relayList = extractRelayList(potentialUserRelays.toVec().first())
+        val relaysToUse =
+            if (relayList.any { (_, relayType) -> relayType == RelayMetadata.WRITE }) {
+                relayList.filter { it.value == RelayMetadata.WRITE }.map { entry -> entry.key }
+            } else if (relayList.size < 7) {
+                relayList.map { entry -> entry.key } // This represents the relay URL, just as the operation above.
+            } else {
+                defaultArticleFetchRelays.map { it }
+            }
+
+        return relaysToUse
+    }
+
+    private suspend fun fetchArticlesForAuthor(
+        author: PublicKey,
+        relays: List<String>,
+    ): List<Event> {
+        val articlesByAuthorFilter =
+            Filter()
+                .author(author)
+                .kind(Kind.fromEnum(KindEnum.LongFormTextNote))
+        logDebug(LOG_TAG, "Relay List size: ${relays.size}")
+
+        nostrClient.removeAllRelays()
+        val relaysToUse =
+            relays.take(2).plus(defaultArticleFetchRelays.random())
+                .ifEmpty { defaultFetchRelays }
```

Comment:
and here, is there a specific reason for taking up to 2 and a random other relay?

Reply:
I should probably bump the number to 3 relays, but the reasoning here is similar to the one above, and even more so because the author can have a large number of articles, and connecting to many relays will worsen the fetch time. There's an author who had 272 articles, and it took about 40s to fetch them!